src/storage/cache/rpc/client.rs

```diff
@@ -0,0 +1,660 @@
+    timeout_options: ClientTimeoutOptions,
+    /// Stream auto increment sequence number, used to mark the request and response
+    seq: AtomicU64,
+    /// The instant of the connection, used to calculate the keep alive timeout
+    clock_instant: Instant,
+    /// Received keep alive timestamp, will mark the valid keep alive response.
+    /// If the keep alive response is not received in right mestamp, the connection will be closed,
+    /// If we receive other response, we will update the received_keepalive_timestamp too, just treat it as a keep alive response.
+    received_keepalive_timestamp: AtomicU64,
+    /// Send packet task
+    packets_keeper: PacketsKeeper<P>,
+    /// Response buffer, try to reuse same buffer to reduce memory allocation
+    /// In case of the response is too large, we need to consider the buffer size
+    /// Init size is 4MB
+    /// Besides, we want to reduce memory copy, and read/write the response to the same data buffer
+    resp_buf: UnsafeCell<BytesMut>,
+    /// Request encode buffer
+    req_buf: UnsafeCell<BytesMut>,
+    /// The Client ID for the connection
+    client_id: u64,
+}
+
+// TODO: Add markable id for this client
+impl<P> Debug for RpcClientConnectionInner<P>
+where
+    P: Packet + Clone + Send + Sync + 'static,
+{
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("RpcClientConnectionInner")
+            .field("client id", &self.client_id)
+            .finish_non_exhaustive()
+    }
+}
+
+impl<P> RpcClientConnectionInner<P>
+where
+    P: Packet + Clone + Send + Sync + 'static,
+{
+    /// Create a new connection.
+    pub fn new(stream: TcpStream, timeout_options: &ClientTimeoutOptions, client_id: u64) -> Self {
+        Self {
+            stream: UnsafeCell::new(stream),
+            timeout_options: timeout_options.clone(),
+            seq: AtomicU64::new(0),
+            clock_instant: Instant::now(),
+            received_keepalive_timestamp: AtomicU64::new(0),
+            packets_keeper: PacketsKeeper::new(timeout_options.clone().task_timeout.as_secs()),
+            resp_buf: UnsafeCell::new(BytesMut::with_capacity(8 * 1024 * 1024)),
+            req_buf: UnsafeCell::new(BytesMut::with_capacity(8 * 1024 * 1024)),
+            client_id,
+        }
+    }
+
+    /// Recv request header from the stream
+    pub async fn recv_header(&self) -> Result<RespHeader, RpcError> {
+        // Try to read to buffer
+        match self.recv_len(REQ_HEADER_SIZE).await {
+            Ok(()) => {}
+            Err(err) => {
+                debug!("Failed to receive request header: {:?}", err);
+                return Err(err);
+            }
+        }
+
+        let buffer: &mut BytesMut = unsafe { &mut *self.resp_buf.get() };
+        let req_header = RespHeader::decode(buffer)?;
+        debug!("Received response header: {:?}", req_header);
+
+        Ok(req_header)
+    }
+
+    /// Receive request body from the server.
```

Comment:
Is the comment correct?

Reply:
It should be response body, I will modify it.